File: guides/micronaut-integration-testing/java/src/main/java/example/micronaut/HelloRepositoryDefault.java
package example.micronaut;

import jakarta.inject.Singleton;

@Singleton // <1>
class HelloRepositoryDefault implements HelloRepository {

    HelloRepositoryDefault() {}

    @Override
    public String findHelloByLanguage(String language) {
        return "Could not hello :(";
    }

    @Override
    public void putHelloInLanguage(String language, String hello) {
    }
}

File: guides/micronaut-integration-testing/kotlin/src/main/kotlin/example/micronaut/HelloRepositoryDefault.kt
package example.micronaut

import jakarta.inject.Singleton

@Singleton // <1>
class HelloRepositoryDefault: HelloRepository {

    override fun findHelloByLanguage(language: String): String {
        return "Could not hello :("
    }

    override fun putHelloInLanguage(language: String, hello: String) {}
}